internal/pkg/ntp/ntp_test.go

```diff
@@ -212,6 +241,57 @@ func (suite *NTPSuite) TestSyncContinuous() {
 	wg.Wait()
 }
 
+func (suite *NTPSuite) TestSyncWithSpikes() {
+	syncer := ntp.NewSyncer(logging.Wrap(log.Writer()).With(zap.String("controller", "ntp")), []string{"127.0.0.7"})
+
+	syncer.AdjustTime = suite.adjustSystemClock
+	syncer.CurrentTime = suite.getSystemClock
+	syncer.NTPQuery = suite.fakeQuery
+
+	syncer.MinPoll = time.Second
+	syncer.MaxPoll = time.Second
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		syncer.Run(ctx)
+	}()
+
+	select {
+	case <-syncer.Synced():
+	case <-time.After(10 * time.Second):
+		suite.Assert().Fail("time sync timeout")
+	}
+
+	suite.Assert().NoError(
+		retry.Constant(12*time.Second, retry.WithUnits(100*time.Millisecond)).Retry(func() error {
+			suite.clockLock.Lock()
+			defer suite.clockLock.Unlock()
+
+			if len(suite.clockAdjustments) < 6 {
+				return retry.ExpectedError(fmt.Errorf("not enough syncs"))
+			}
+
+			for _, adj := range suite.clockAdjustments {
+				// 1s spike should be filtered out
+				suite.Assert().Equal(time.Millisecond, adj)
+			}
+
+			return nil
+		}))
```

Comment:
nitpick: I think the last `)` should be on the next line

Reply:
fair enough!